multi-educator/src/Dscp/MultiEducator/Launcher/Mode.hs

```diff
@@ -157,7 +157,6 @@ loadEducator login mpassphrase = do
             & sub #witness .~ multiEducatorConfig ^. sub #witness
             & sub #educator . sub #db .~ multiEducatorConfig ^. sub #educator . sub #db
             & sub #educator . sub #keys . sub #keyParams .~ keyParams
-            & sub #educator . sub #api .~ multiEducatorConfig ^. sub #educator . sub #api
```

Comment:
Even tho `newCfg` seems to be not really used (before DSCP-445 the value was `error ""`) the implementation I left actually works and produces a correct (finalized) Educator config because it substitutes **every** `sub`config with one from the (finalized) MultiEducator config we have.

Given that it starts from a `finaliseDeferredUnsafe mempty` if a `sub` substitution gets removed we'll not get a finalized Educator config anymore as our result.

One way to still keep this working would be to define something like:
```
meWebConfig = multiEducatorConfig ^. sub #educator . sub #api
toEducatorNoAuth = \case
    NoAuthOnContext _ -> NoAuthOnContext ()
    NoAuthOffContext -> NoAuthOffContext
```
and replace this removed line with:
```
& sub #educator . sub #api . sub #serverParams .~ meWebConfig ^. sub #serverParams
& sub #educator . sub #api . sub #botConfig .~ meWebConfig ^. sub #botConfig
& sub #educator . sub #api . option #educatorAPINoAuth .~ toEducatorNoAuth (meWebConfig ^. option #multiEducatorAPINoAuth)
& sub #educator . sub #api . option #studentAPINoAuth .~ meWebConfig ^. option #studentAPINoAuth
```
or something like this.

As I said I'm not really sure how important this is and maybe I should have put a comment about the fact that this is basically the creation of a finalized config from another finalized one.

Reply:
Well, that makes sense.
On the other hand, we know that this parameter cannot and shouldn't be used by educator because multi-educator code handles bringing educator API server up. Using `toEducatorNoAuth` would look unpleasant IMO because `NoAuthContext "educator"` and `NoAuthContext "multi-educator"` have pretty different semantics, which may express in us getting stuck if `NoAuthOnContext` for `"educator"` one day contains something smarter than `()`.

I guess I will set it as
`sub #educator . sub #api . option #educatorAPINoAuth .~ error "Do not touch, multi-educator has already run the server"`. So that `finaliseDeferredUnsafe mempty` is again justified (we can replace it with `finalise` with the same effect), but we know that `"educatorAPINoAuth"` config parameter shouldn't be used by the educator.

Reply:
Yeah, that's better!